File: bridge/c-sharp/TextMessageSender.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Bridge
{
    // UML diyagramındaki ConcreteImplementor (TextMessageSender) kısmına denk gelmektedir.
    // Implementor arayüzünü (IMessageSender) uygular.
    public class TextMessageSender : IMessageSender
    {
        public void SendMessage(Message message)
        {
            Console.WriteLine("TextMessageSender: Sending text message...");
        }
    }
}

File: bridge/typescript/text-message-sender.ts
import {IMessageSender} from "./message-sender.interface";
import {Message} from "./message";

// UML diyagramındaki ConcreteImplementor (TextMessageSender) kısmına denk gelmektedir.
// Implementor arayüzünü (IMessageSender) uygular.
export class TextMessageSender implements IMessageSender {
  public sendMessage(message: Message): void {
    console.log('TextMessageSender: Sending text message...');
  }
}
